source/tests/unittests/TestHelixClass.cpp

```diff
@@ -47,3 +49,214 @@ TEMPLATE_LIST_TEST_CASE("Initialize_VP", "[helix-init]", HelixTypes) {
 // - Other usage examples as found in the "real-world"
 // - Also add things that work slightly unexpectedly at the moment, e.g. things
 // that are listed in https://github.com/iLCSoft/MarlinUtil/issues/24
+
+
+TEMPLATE_LIST_TEST_CASE("getDistance-Initialize_VP",
+                        "[helix-init][helix-distance]", HelixTypes) {
+  TestType helix1, helix2;
+  using FloatT = typename TestType::float_type;
+  /*const*/ std::array<FloatT, 3> momentum1 = {10., 20., 30.};
+  /*const*/ std::array<FloatT, 3> momentum2 = {70., 30., 10.};
+  /*const*/ std::array<FloatT, 3> momentum;
+  for (int i=0; i<3; i++) momentum[i] = momentum1[i] + momentum2[i];
+
+  const auto magField = 3.0;
+  const auto charge1 = -1.0;
+  const auto charge2 = 1.0;
+
+  FloatT vtx_momentum1[3];
+  FloatT vtx_momentum2[3];
+  FloatT vertex1[3];
+  FloatT vertex2[3];
+
+  SECTION("zero reference point") {
+    // use ref. point in a "true" vertex for simplicity
+    /*const*/ std::array<FloatT, 3> position = {0, 0, 0};
+
+    helix1.Initialize_VP(position.data(), momentum1.data(), charge1, magField);
+    helix2.Initialize_VP(position.data(), momentum2.data(), charge2, magField);
+
+    FloatT dist1 = helix1.getDistanceToHelix(&helix2, vertex1, vtx_momentum1);
+    FloatT dist2 = helix2.getDistanceToHelix(&helix1, vertex2, vtx_momentum2);
+
+    FloatT eps = 0.1;
+
+    // distance should be 0 as we initialized helices at the same point
+    CHECK_THAT(0.0, Catch::Matchers::WithinAbs(dist1, eps));
+    CHECK_THAT(0.0, Catch::Matchers::WithinAbs(dist2, eps));
```

Comment:
```suggestion
    REQUIRE(dist1 == Catch::Approx(0.0));
    REQUIRE(dist2 == Catch::Approx(0.0));
```
I would have done all the checks like this, as it is slightly easier to read and the `Catch::Approx` is the more robust comparison for floating point numbers compared to `Catch::Matchers::WitihnAbs` (e.g. it will will also take into account the magnitue of the numbers you are comparing). Additionally, this is what I am used to read ;)

Reply:
I've been using Catch::Matchers::WithinAbs just based on the documentation: https://github.com/catchorg/Catch2/blob/devel/docs/comparing-floating-point-numbers.md#approx But if you have good experience with `Catch::Approx` and you prefer it this way, it's completely fine for me so I will replace it.

Reply:
Aha. Thanks for pointing that out. I wasn't aware of this... (Also because it seems like a rather new development). In this case we obviously go with the recommendation of the documentation.